.circleci/deploy-to-pantheon.sh

```diff
@@ -23,7 +29,7 @@ then
     exit 0;
 fi
 
-if -n ${CIRCLE_PULL_REQUEST+x} && ! TERMINUS_DOES_MULTIDEV_EXIST ${TERMINUS_ENV}
```

Comment:
`-if -n ${CIRCLE_PULL_REQUEST+x}` was returning `./.circleci/deploy-to-pantheon.sh: line 26: conditional binary operator expected` errors in Circle CI.

Having `TERMINUS_DOES_MULTIDEV_EXIST` return `1` when on the `master` branch eliminates the need for that part of the conditional.